### PR TITLE
fix negative number parsing for positions

### PR DESCRIPTION
**Pull request by Machine-Maker**
Parsing of positions failed to account for negative numbers which according to Minecraft's parsing of these positions should be accounted for.

### CODE REVIEW

File: api/src/test/java/net/kyori/adventure/text/BlockNBTComponentTest.java [hunk 1]
```diff
@@ -95,16 +95,16 @@ void testLocalPosParsing() {
...
   }
 
   @Test
   void testRelativeWorldPosParsing() {
     assertEquals(
-      BlockNBTComponent.WorldPos.worldPos(BlockNBTComponent.WorldPos.Coordinate.relative(7), BlockNBTComponent.WorldPos.Coordinate.relative(83), BlockNBTComponent.WorldPos.Coordinate.relative(900)),
-      BlockNBTComponent.Pos.fromString("~7 ~83 ~900")
+      BlockNBTComponent.WorldPos.worldPos(BlockNBTComponent.WorldPos.Coordinate.relative(-7), BlockNBTComponent.WorldPos.Coordinate.relative(83), BlockNBTComponent.WorldPos.Coordinate.relative(900)),
+      BlockNBTComponent.Pos.fromString("~-7 ~83 ~900")
     );
```
Comment (kashike) [inline api/src/test/java/net/kyori/adventure/text/BlockNBTComponentTest.java]: Please don't modify the tests to change the numbers - add new tests for negative numbers.

**Review by kezz — APPROVED** [review]

**Review by kashike — APPROVED** [review]
